crates/store/src/state.rs

```diff
@@ -332,6 +327,16 @@ impl State {
             // successfully.
             let mut inner = self.inner.write().await;
 
+            // We need to check that neither the nullifier tree root, nor the account tree root
+            // haven't changed while we were waiting for the DB preparation task to complete.
+            // If it has changed, we mustn't proceed with both in-memory and database updates,
+            // since it might lead to inconsistent state.
```

Comment:
nit: I would write this as:
```rust
// We need to check that neither the nullifier tree nor the account tree have changed
// while we were waiting for the DB preparation task to complete. If either of them 
// did changed, we do not proceed with in-memory and database updates, since it may
// lead to an inconsistent state.
```